src/libfetchers/fetch-to-store.cc

```diff
@@ -13,8 +14,16 @@ StorePath fetchToStore(
     PathFilter * filter,
     RepairFlag repair)
 {
-    // FIXME: add an optimisation for the case where the accessor is
-    // an FSInputAccessor pointing to a store path.
+    if (path.accessor->isStorePath
+        && path.path.isRoot()
+        && method == FileIngestionMethod::Recursive
```

Comment:
What goes wrong if we skip this? I am trying to special-case NAR hashing *less* in Nix currently.
```suggestion
```